lib/sbi_ipi.c

```diff
@@ -151,10 +164,15 @@ void sbi_ipi_process(struct sbi_scratch *scratch)
 			__asm__ __volatile("fence.i");
 			break;
 		case SBI_IPI_EVENT_SFENCE_VMA:
-			sbi_ipi_sfence_vma(sbi_tlb_info_ptr(scratch));
-			break;
 		case SBI_IPI_EVENT_SFENCE_VMA_ASID:
-			sbi_ipi_sfence_vma_asid(sbi_tlb_info_ptr(scratch));
+			tinfo = sbi_fifo_dequeue(ipi_tlb_fifo);
+			while (tinfo) {
+				if (tinfo->type == SBI_TLB_FLUSH_VMA)
+					sbi_ipi_sfence_vma(tinfo);
+				else if (tinfo->type == SBI_TLB_FLUSH_VMA_ASID)
+					sbi_ipi_sfence_vma_asid(tinfo);
+				tinfo = sbi_fifo_dequeue(ipi_tlb_fifo);
+			}
```

Comment:
Similar to above, instead of duplicating the call to `sbi_fifo_dequeue()`, I think the while loop can be better written as:

```
while (sbi_fifo_dequeue(ipi_tlb_fifo)) {
    if condition
}
```